### PR TITLE
Background sync iOS docs

### LINKED ISSUE #3785: Write documentation describing how the background code execution works on iOS
Write down all the necessary information about how the scheduling of background app refresh works and add links to important parts of the official Apple documentation (or to Xamarin docs if necessary).

List all the conditions that have to be met so that our code is run by the operating system. Is it possible that the app and the background refresh run at the same time? How do we deal with that?

Make the docs as detailed as possible. Add it into a separate iOS-only `md` file in the `docs/syncing` directory.

### PR DESCRIPTION
**Pull request by dshr**
## What's this?
Documents our background sync process, and goes a bit further in-depth re the iOS implementation.

### Relationships
Closes #3785

## Review hints
Check if it's all easily readable, I struggled with that a lot..

## :squid: Permissions
Sync ppl please.

### CODE REVIEW

File: docs/syncing/bg-fetch-ios.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Background Fetch on iOS
+=======================
+
+To enable background syncing we use the iOS [Background Fetch API](https://developer.apple.com/documentation/uikit/core_app/managing_your_app_s_life_cycle/preparing_your_app_to_run_in_the_background/updating_your_app_with_background_app_refresh). In our `BaseBackgroundSyncService` we subscribe to Login and Logout events, while in its subclass`BackgroundSyncServiceIos` we have the event callbacks.  In them we enable or disable fetching using the `SetMinimumBackgroundFetchInterval` method of `UIApplication`. We either set the interval to 15 minutes (as defined in `BackgroundSyncServiceIos`), or to the constant `UIApplication.BackgroundFetchIntervalNever` to turn fetching on and off respectively.
+
+What happens then is that whenever the app is in the background and we have fetching enables, iOS will try and find some time to wake our app and call it's `PerformFetch` handler. The handler supplies us with a `UIApplication` reference and a `Action<UIBackgroundFetchResult>` completion handler callback. We have to call this handler with the appropriate `UIBackgroundFetchResult` when our background fetch is done.
```
Comment (Odrakir) [inline docs/syncing/bg-fetch-ios.md]:
```suggestion
What happens then is that whenever the app is in the background and we have fetching enabled, iOS will try and find some time to wake our app and call it's `PerformFetch` handler. The handler supplies us with a `UIApplication` reference and a `Action<UIBackgroundFetchResult>` completion handler callback. We have to call this handler with the appropriate `UIBackgroundFetchResult` when our background fetch is done. 
```

Comment (Odrakir) [inline docs/syncing/bg-fetch-ios.md]:
```suggestion
What happens then is that whenever the app is in the background and we have fetching enables, iOS will try and find some time to wake our app and call its `PerformFetch` handler. The handler supplies us with a `UIApplication` reference and a `Action<UIBackgroundFetchResult>` completion handler callback. We have to call this handler with the appropriate `UIBackgroundFetchResult` when our background fetch is done. 
```

File: docs/syncing/bg-sync-switching.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Background sync switching
```
Comment (Odrakir) [inline docs/syncing/bg-sync-switching.md]: scheduling?

Comment (simonrozsival) [reply docs/syncing/bg-sync-switching.md]: I agree that this section's name is a bit confusing. I would call it "Enabling and disabling background syncing".

**Pull request merged**